src/analysisd/analysisd.c

```diff
@@ -1661,8 +1661,11 @@ RuleInfo *OS_CheckIfRuleMatch(Eventinfo *lf, RuleNode *curr_node)
     /* If it is a context rule, search for it */
     if(currently_rule->context == 1)
     {
-        if(!currently_rule->event_search(lf, currently_rule))
-            return(NULL);
+
+	if(!(currently_rule->context_opts & SAME_DODIFF)) {
+        	if(!currently_rule->event_search(lf, currently_rule))
+            	return(NULL);
```

Comment:
i am being to pedantic in that this indention is very unclear.   Going to pull this in, as we have not formalized the coding sytles stuff yet.